virttest/utils_test/__init__.py

```diff
@@ -2318,26 +2328,179 @@ def unload_stress(stress_type, params, vms=None, remote_server=False):
+        self.client_vms = []
+        self.server_vms = []
+        if (self.need_profile):
+            self.profile = params.get("client_profile_%s" % self.stress_type)
+            self.profile = os.path.join(data_dir.get_root_dir(), self.profile)
+            self.profile_pattren = params.get("profile_pattren", "").split()
+
+        for index, vm in enumerate(self.vms):
+            if index % 2 != 0:
+                # Process for client
+                self.client_vms.append(vm)
+            else:
+                # Process for server
+                self.server_vms.append(vm)
+        if len(self.server_vms) != len(self.client_vms):
+            test.cancel(
+                "This test requires server and client vms in 1:1 ratio")
+        self.pair_vms = zip(self.server_vms, self.client_vms)
+
+    def prepare_profile(self, test, fpath, pat_repl):
+        """
+        This is to prepare client profile to be run on client.
+        :param test: kvm test object
+        :param fpath: profile to be run
+        :param pat_repl: dict containing pattern and replacement : includes server_ip, duration, threads etc.
+        :return: no explicit return. But profile on fpath would be edited and ready to be run on client guest
+        :raise: TestError: raised if unable to edit the given profile
+        """
+        try:
+            if not self.profile.endswith(".xml"):
+                logging.debug("Error: profile should be an xml: %s", self.profile)
+                test.cancel("%s profile not valid", self.stress_type)
+            with open(fpath, 'r+') as profile_content:
+                tempstr = profile_content.read()
+                profile_content.truncate(0)
+                logging.debug(
+                    "In prepare profile: pattern and replacement : %s", pat_repl)
+                for pattern, replace in pat_repl.items():
+                    tempstr = tempstr.replace(pattern, replace)
+                profile_content.write(tempstr)
+            logging.debug("Profile xml to be run : %s ", tempstr)
+        except Exception:
+            test.error("Failed to update file : %s", fpath)
+
+    def load_stress(self, test, params):
+        """
+        This function can be called to load server client stress tools into
+        vms/baremetal hosts.
+        : test: kvm test Object
+        :param vms: list of vms [should be in pairs]
+        :param params: test params for stress tools
+        :return: Based on result of test would return True or False
+        :raise: TestError: raised if unable load stress tool or given input is incorrect
+        """
+
+        error = False
+        for server_vm, client_vm in self.pair_vms:
+            try:
+                params['stress_cmds_%s' % self.stress_type] = self.server_cmd
+                stress_server = VMStress(server_vm, self.stress_type, params)
+                params['server_pwd'] = params.get("password")
```

Comment:
we currently use same guest password on all our guests.

Reply:
It makes sense to use `server_params = params.object_params(server_vm)`, it will help us not only to have different password for different VM also to configure iptable table rules or any VM specific configurations in different client-server tool. similar to in line L2520, as your patch is designed in generic way,  let's make use of `params.object_params()`